util/sync_dumps_and_logs

```diff
@@ -0,0 +1,133 @@
+#
+# Well, glad you asked!
+#
+# Simply put, it is a set of sed commands to read the comments above this one,
+# and print it out, minus the comment characters and first whitespace char,
+# which will then produce our help info.
+# 
+# Specifically, in order, each of the these `-e` args are a seperate command to
+# execute, in order, for each line of this file (though, this never really ends
+# up getting to the end of the file... more on that below):
+#
+#    * '1,2d' will run the 'd' command on the first two lines when parsing this
+#      file.  That will simply no print that output, skip running any other
+#      '-e' commands, and start on the next line
+#    * '3,$ s/^# \{0,1\}\(.*\)$/\1/' simply matches any comment char until the
+#      end of the file, and prints the line, minus the '#' and the first
+#      whitespace char following (if it exists).  The blank lines are desired
+#      for formatting, hence the `\{0,1\}` bit.
+#    * 'tx` is a jump statement, and will execute if there was a match on this
+#      line in anything preceeding it.  If not, it is effectively a no-op.  If
+#      there was a match in one of the previous commands, it will jump to the
+#      ':x' command.
+#    * 'c\' Since there wasn't a match with this line, don't print, at least for
+#      our purposes (this is a bit of a hack, since 'd' would cause the next
+#      line to never get triggered).
+#    * 'q' and quits... duh.  That said, the ':x' in the next command, and the
+#      previous '1,2d' commands are important, because they avoid this command
+#      since they either start the execution of all the commands over on the
+#      next line, or are after this one.
+#
+# The 'q' is necessary, instead of another 'd', because this comment or others
+# are not wanted in the help output for the command, and are just reference for
+# developers.  So the help docs at the top of the file just need to be
+# seperated by a single blank line, or some code that isn't a comment, and then
+# that is where they stop when being printed via the '-h' flag.
+#
+# There you have it. TMYK!   (ﾉ☉ヮ⚆)ﾉ ⌒*:･ﾟ✧
+#
+help() {
+  sed -e'1,2d' -e'3,$ s/^# \{0,1\}\(.*\)$/\1/' -e'tx' -e'c\' -e'q' -e':x' $0
+}
+
+NextDateStamp=$(date -v+1d "+%Y%m%d")
+
+RemoteUser=root
+RemotePass=smartvm
+LocalLogDir=tmp/mem_prof_logs
+LocalDumpDir=tmp/mem_prof_dumps
+DryRun=""
+
+while getopts "h?H:U:P:d:l:ir" opt; do
+  case "$opt" in
+  h) help
+     exit 0
+     ;;  
+  H) RemoteHost=${OPTARG} ;; 
+  U) RemoteUser=${OPTARG} ;;
+  P) RemotePass=${OPTARG} ;;
+  d) LocalDumpDir=${OPTARG} ;;
+  l) LocalLogDir=${OPTARG} ;;
+  i) IpDirs="1" ;;
+  r) DryRun="--dry-run -v" ;;
+  esac
+done
+
+if [[ "$IpDirs" == "1" ]]; then
+  IpDir=$(echo $RemoteHost | tr "." "-")
+  LocalLogDir="$LocalLogDir/$IpDir"
+  LocalDumpDir="$LocalDumpDir/$IpDir"
+fi
+
+mkdir -p $LocalLogDir
+mkdir -p $LocalDumpDir
+
+run_ssh_cmd() {
+  expect -c "
+  set timeout -1
```

Comment:
This bit is important by the way (probably should add a comment here).  `expect`, by default, will timeout if the command doesn't do anything within 15 seconds.  Since the `gzip`'ing of the dump files can take a while (since I didn't have that automated besides this script), this is needed to give the first `ssh` command time to finish.